src/mgmt/srv_drpc.c

```diff
@@ -801,116 +870,189 @@ void ds_mgmt_drpc_pool_set_prop(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 		D_GOTO(out, rc = -DER_INVAL);
 	}
 
-	D_INFO(DF_UUID": received request to set pool property\n",
+	D_INFO(DF_UUID": received request to set pool properties\n",
 	       DP_UUID(uuid));
 
-	new_prop = daos_prop_alloc(1);
-	if (new_prop == NULL) {
-		D_ERROR("Failed to allocate daos property\n");
+	rc = conv_req_props(&new_props, true,
+			    req->properties, req->n_properties);
+	if (rc != 0)
+		goto out;
+
+	svc_ranks = uint32_array_to_rank_list(req->svc_ranks, req->n_svc_ranks);
+	if (svc_ranks == NULL)
 		D_GOTO(out, rc = -DER_NOMEM);
+
+	rc = ds_mgmt_pool_set_prop(uuid, svc_ranks, new_props);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to set pool properties: "DF_RC"\n",
+			DP_UUID(uuid), DP_RC(rc));
+		goto out_ranks;
 	}
 
-	if (req->property_case != MGMT__POOL_SET_PROP_REQ__PROPERTY_NUMBER) {
-		D_ERROR("Pool property request must be numeric\n");
-		D_GOTO(out, rc = -DER_INVAL);
+out_ranks:
+	d_rank_list_free(svc_ranks);
+out:
+	daos_prop_free(new_props);
+
+	resp.status = rc;
+	len = mgmt__pool_set_prop_resp__get_packed_size(&resp);
+	D_ALLOC(body, len);
+	if (body == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILED_MARSHAL;
+		D_ERROR("Failed to allocate drpc response body\n");
```

Comment:
I've removed a couple of these elsewhere but please don't log D_ALLOC() failures, the macro will do that for you.

Reply:
Ah, ok. Will remove those.